receiver/datadogreceiver/receiver.go

```diff
@@ -35,8 +36,9 @@ type datadogReceiver struct {
 	shutdownWG   sync.WaitGroup
 	tReceiver    *obsreport.Receiver
 
-	startOnce sync.Once
-	stopOnce  sync.Once
+	startOnce    sync.Once
+	stopOnce     sync.Once
+	errorCounter atomic.Uint64
```

Comment:
Sorry, I should have been clearer: I meant that it would be useful to have a metric (counter) here, so that people managing the collector could monitor the error count and react in case of problems.

Reply:
Hahaha so confused dude, either way we need the atomic error counter. Do you know an example of exporting a metric from a receiver I can yoink?

Reply:
@jpkrohling I found this mdatagen stuff, give it a look

Reply:
That's for consuming metrics datapoints in receivers. I'm thinking something more like this, in order to observe the component itself:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/loadbalancingexporter/metrics.go

Or perhaps the obsreport can be used? https://github.com/open-telemetry/opentelemetry-collector/tree/main/obsreport